### PR TITLE
Moving stats manager and related classes to com.github.ambry.server

### PR DESCRIPTION
**Pull request by vgkholla**
Moving StatsManager, tests and the related avro classes to com.github.ambry.server

### CODE REVIEW

File: ambry-server/src/integration-test/java/com.github.ambry.server/StatsManagerTest.java [hunk 1]
```diff
@@ -258,37 +278,84 @@ public Store getStore(PartitionId id) {
   }
 
   /**
-   * Mocked {@link BlobStore} that is intended to return a predefined {@link StoreStats} when getStoreStats is called.
+   * Mocked {@link Store} that is intended to return a predefined {@link StoreStats} when getStoreStats is called.
    */
-  private class MockBlobStore extends BlobStore {
+  private class MockBlobStore implements Store {
```
Comment (xunyin8) [inline ambry-server/src/integration-test/java/com.github.ambry.server/StatsManagerTest.java]: maybe MockStore instead.

File: ambry-server/src/integration-test/java/com.github.ambry.server/StatsManagerTest.java [hunk 2]
```diff
@@ -258,37 +278,84 @@ public Store getStore(PartitionId id) {
...
+      throw new IllegalStateException("Not implemented");
+    }
   }
 
   /**
-   * Mocked {@link BlobStoreStats} to return predefined {@link StatsSnapshot} when getStatsSnapshot is called.
+   * Mocked {@link StoreStats} to return predefined {@link StatsSnapshot} when getStatsSnapshot is called.
    */
-  private class MockBlobStoreStats extends BlobStoreStats {
+  private class MockBlobStoreStats implements StoreStats {
```
Comment (xunyin8) [inline ambry-server/src/integration-test/java/com.github.ambry.server/StatsManagerTest.java]: Similarly, MockStoreStats instead.

**Comment by vgkholla** [comment]
@xunyin8 Addressed comments.

**Comment by vgkholla** [comment]
Also added some logging messages and moved some configs around.

File: ambry-server/src/main/java/com.github.ambry.server/StatsManagerMetrics.java [hunk 3]
```diff
@@ -12,23 +12,26 @@
...
 
+/**
+ * Metrics for {@link StatsManager} and related components.
+ */
 class StatsManagerMetrics {
   final Counter statsPublishFailureCount;
-  final Histogram totalFetchAndAggregateTime;
-  final Histogram fetchAndAggregateTimePerStore;
+  final Histogram totalFetchAndAggregateTimeMs;
+  final Histogram fetchAndAggregateTimeMsPerStore;
```
Comment (nsivabalan) [inline ambry-server/src/main/java/com.github.ambry.server/StatsManagerMetrics.java]: fetchAndAggregateTimePerStoreMs

File: ambry-api/src/main/java/com.github.ambry/config/StatsManagerConfig.java [hunk 4]
```diff
@@ -34,15 +34,15 @@
   public final long publishPeriodInSecs;
 
   /**
-   * The option to enable or disable publishing stats.
+   * The time period in seconds to delay before the first stats collection is triggered
```
Comment (xunyin8) [inline ambry-api/src/main/java/com.github.ambry/config/StatsManagerConfig.java]: Might be good to mention it in the comment or rename to initial.delay.upper.bound.in.secs to reflect the fact that the actual delay is randomly chosen between 0 and this value (exclusive).

File: ambry-server/src/integration-test/java/com.github.ambry.server/StatsManagerTest.java [hunk 5]
```diff
@@ -65,22 +79,27 @@
    */
   @After
   public void cleanup() throws IOException {
-    assertTrue(tempDir.getAbsolutePath() + " could not be deleted", StoreTestUtils.cleanDirectory(tempDir, true));
+    File[] files = tempDir.listFiles();
```
Comment (nsivabalan) [inline ambry-server/src/integration-test/java/com.github.ambry.server/StatsManagerTest.java]: can you move this file to "test" directory. As of now it is in "integration-test".

File: ambry-server/src/main/java/com.github.ambry.server/StatsManagerMetrics.java (hunk 3 above)
Comment (vgkholla) [inline ambry-server/src/main/java/com.github.ambry.server/StatsManagerMetrics.java]: Done.

File: ambry-api/src/main/java/com.github.ambry/config/StatsManagerConfig.java (hunk 4 above)
Comment (vgkholla) [inline ambry-api/src/main/java/com.github.ambry/config/StatsManagerConfig.java]: Done.

File: ambry-server/src/integration-test/java/com.github.ambry.server/StatsManagerTest.java (hunk 5 above)
Comment (vgkholla) [inline ambry-server/src/integration-test/java/com.github.ambry.server/StatsManagerTest.java]: Done

**Comment by vgkholla** [comment]
@xunyin8 @nsivabalan Comments addressed. I have also expanded the try catch in the stats aggregation job. Please take a look

**Comment by vgkholla** [comment]
Formatted and built on mac/Linux

**Comment by xunyin8** [comment]
reviewed the latest changes, LGTM.

**Comment by nsivabalan** [comment]
Looks good. Merging.

**Pull request merged**